### PR TITLE
local quorum using raft with 3 nodes and proxy in docker containers

### PR DESCRIPTION
**Pull request by MajdT51**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,120 @@
+# The following environment variables are substituted if present
+# * QUORUM_CONSENSUS: raft
```
Comment (jpbu) [inline docker-compose.yml]: This variable is no longer used and should be removed from the comment

File: README.md [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+# quorum-local-raft-network
+
+Work In Progress
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+Setup a local quroum network with 3 nodes using raft consensus algorithm without tx manager.
+
+## Rpc proxy to avoid timestamp error
+Raft use the timestamp in nano seconds which is bigger than the max number in javascript,
```
Comment (jpbu) [inline README.md]: Should be explained in a bit more detail, or linked to some background information - e.g. what does this mean, can I use it at all without or proxy or which apps or libraries does this affect?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  
+# use alpine package manager to install git
+RUN apk add --no-cache git
+
+# clone the proxy repository
+RUN mkdir -p /usr/src/app
+
+WORKDIR /usr/src/app
+
+RUN git clone https://github.com/MajdT51/quorum-rpc-proxy.git
```
Comment (jpbu) [inline Dockerfile]: Looks like your PR was merged already and we can use the original repository

File: README.md [hunk 4]
```diff
@@ -1,5 +1,55 @@
...
+change to the example directory:  `cd ./example`
+
+### With Proxy
+to deploy contract and library with proxy: `truffle migrate --reset --network nodeoneproxy`
+
+to test the contract with proxy: `truffle test ./test/metacoin.js --network nodeoneproxy`
+
+### Without Proxy 
+the given truffle commands will start the process,
+but then it will either give an error back or get stuck without any information.
```
Comment (jpbu) [inline README.md]: Do we really need this then? If I understand correctly, it does not work

File: example/contracts/ConvertLib.sol [hunk 5]
```diff
@@ -0,0 +1,8 @@
+pragma solidity >=0.4.25 <0.6.0;
+
+library ConvertLib{
```
Comment (jpbu) [inline example/contracts/ConvertLib.sol]: Use proper Solidity formatting (space before brackets and commas)

File: example/test/metacoin.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+const MetaCoin = artifacts.require("MetaCoin");
```
Comment (jpbu) [inline example/test/metacoin.js]: Either use single quotes or double quotes for strings, but not mixed

**Review by jpbu — COMMENTED** [review]
I tested the setup on my machine and it worked.

General comment: I would not add the `/example` directly in the source code, but rather provide a documentation only in the README how an example truffle config would look like, maybe instructions how to test it with a truffle box. I believe people will be most interested in the minimal local quorum setup and not the MetaCoin. If the example stays in there, the `build` directory should be gitignored.

File: README.md (hunk 4 above)
Comment (MajdT51) [reply README.md]: yes, but they can try it and see that it does not respond correctly

File: Dockerfile (hunk 3 above)
Comment (MajdT51) [reply Dockerfile]: but he maybe could change something in the future?

Comment (jpbu) [reply Dockerfile]: To avoid that, we can checkout a specific version/commit from the original repository

File: README.md [hunk 7]
```diff
@@ -1,5 +1,55 @@
...
+access to geth of nodes: `docker exec -it <container_name> geth attach ./qdata/dd/geth.ipc`
+
+check logs of proxy: `docker logs quorum_raft_proxy_1`
+
+stop:       `docker-compose stop`
+
+remove:     `docker-compose down -v`
+
+## Run truffle example
+based on the meta coin truffle-box example
```
Comment (jpbu) [inline README.md]: Link to truffle / truffle box would be useful

File: example/truffle-config.js [hunk 8]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  networks: {
+     nodeone: {
```
Comment (jpbu) [inline example/truffle-config.js]: Looks like the is not correctly indented

File: README.md [hunk 9]
```diff
@@ -1,5 +1,55 @@
-# quorum-local-raft-network
-
-Work In Progress
-
-Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+# quorum-local-raft-network
+
+Work In Progress
```
Comment (jpbu) [inline README.md]: Is it still work in progress? What is missing?

File: README.md [hunk 10]
```diff
@@ -1,5 +1,24 @@
...
+# quorum-local-raft-network
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+
+Setup a minimal local [quorum network](https://www.goquorum.com/) with 3 nodes using [raft consensus](https://raft.github.io/) algorithm without tx manager (does not support private transactions).
+
+## Quorum local
+Build a local Quorum network based on the [7-nodes example](https://github.com/jpmorganchase/quorum-examples) and simplified to 3 nodes with only [Raft consensus](https://raft.github.io/) without tx manager.
+
+## Run the 3 nodes in containers plus one container for the proxy
```
Comment (jpbu) [inline README.md]: Remove "plus one container for the proxy"

**Pull request merged**